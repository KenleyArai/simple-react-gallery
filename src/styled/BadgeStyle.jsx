import styled from "styled-components";

const BadgeStyle = styled.div``;

export default BadgeStyle;
