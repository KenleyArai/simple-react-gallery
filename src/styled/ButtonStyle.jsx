import styled from "styled-components";

const ButtonStyle = styled.div``;

export default ButtonStyle;
