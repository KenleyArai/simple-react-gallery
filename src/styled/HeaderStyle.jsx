import styled from "styled-components";

const HeaderStyle = styled.span``;

export default HeaderStyle;
