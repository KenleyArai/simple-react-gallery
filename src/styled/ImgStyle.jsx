import styled from "styled-components";

const ImgStyle = styled.img``;

export default ImgStyle;
