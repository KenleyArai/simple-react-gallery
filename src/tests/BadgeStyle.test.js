import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import "jest-styled-components";

configure({ adapter: new Adapter() });

import blank from "../images/blank.svg";
import linkedin from "../images/linkedin.svg";
import facebook from "../images/facebook.svg";

import { background_helper } from "../styled/BadgeStyle";

describe("BadgeStyled", () => {
  describe("Helper Functions", () => {
    describe("background_helper", () => {
      it("should return blank icon when type is blank", () => {
        expect(background_helper("blank")).toBe(blank);
      });
      it("should return linkedin icon when type is linkedin", () => {
        expect(background_helper("linkedin")).toBe(linkedin);
      });
      it("should return facebook icon when type is facebook", () => {
        expect(background_helper("facebook")).toBe(facebook);
      });
    });
  });
});
