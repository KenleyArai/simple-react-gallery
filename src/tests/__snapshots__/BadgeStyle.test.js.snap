// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeStyled should render with a blank background with blank prop 1`] = `
.c0 {
  background-image: blank.svg;
}

<div
  className="c0"
  type="blank"
/>
`;

exports[`BadgeStyled should render with a facebook background with facebook prop 1`] = `
.c0 {
  background-image: facebook.svg;
}

<div
  className="c0"
  type="facebook"
/>
`;

exports[`BadgeStyled should render with a linkedin background with linkedin prop 1`] = `
.c0 {
  background-image: linkedin.svg;
}

<div
  className="c0"
  type="linkedin"
/>
`;
